src/main/java/org/littleshoot/proxy/mitm/SubjectAlternativeNameHolder.java

```diff
@@ -47,8 +47,8 @@ public void addAll(Collection<List<?>> subjectAlternativeNames) {
     }
 
     private ASN1Encodable parseGeneralName(List<?> nameEntry) {
-        if (nameEntry != null && nameEntry.size() != 2) {
-            throw new IllegalArgumentException(String.valueOf(nameEntry));
+        if (nameEntry == null || nameEntry.size() != 2) {
```

Comment:
It was the best solution I think because in case of two if blocks both conditions would have to evaluate to true for body execution and in this case either nameEntry is null or nameEntry.size() is not 2 body should be executed. If you have any better suggestions please let me know with a little example.